### PR TITLE
Added sprint goal

### PR DESCRIPTION
**Pull request by raya311**
# Pull Request

## Summary

Added spring planning/goal document. Documented as per 6th April meeting.

## Why

To better deliver the software and distribute the work.

## How to Verify

Provide clear steps to verify this change locally.

1. Install dependencies: `npm install`
2. Run the app / relevant command
3. Run lint: `npm run eslint`
4. Run tests: `npm run test`
5. Describe expected behavior

## Checklist (Definition of Done)

- Branch follows team naming pattern (`week<NUM>/<short-description>`)
- PR opened early and kept reviewable in size
- At least one teammate review requested
- CI checks are passing
- Lint passes locally (`npm run eslint`)
- Tests pass locally (`npm run test`)
- New behavior includes starter tests or test plan notes
- Documentation updated if needed (`README`, `docs/`, ADRs)

## Notes for Reviewers

Anything specific you'd like feedback on?
- Edge cases?
- Architecture concerns?
- Naming?
- Test coverage?

### CODE REVIEW

File: docs/final/week13-sprint.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+| # | Item | Owner | Priority |
+|---|------|-------|----------|
+| 1 | Enhance error handling and user feedback in API failures | Jason | 2 |
+| 2 | Architecture update snapshot | Jason | 2 |
+| 3 | Flaky edit-mode UI after loading saved trip (context/button state race) | Heather | 2 |
+| 4 | Duplicated trip payload validation across create/update routes | Heather | 2 |
+| 5 | Hand off documentation | Heather | 3 |
+| 6 | Add a unit/integration test around change detection | Naren | 2 |
+| 7 | Add Playwright regression test for authError redirect handling | Naren | 2 |
```
Comment (hmhawn) [inline docs/final/week13-sprint.md:19]: items 7 and 8 are almost identical?

File: docs/final/week13-sprint.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+- Stack and tool choices
+- Setup/run summary
+- Known weaknesses or technical debt
+- Recommended next steps for a future team
+
+---
+
+### 6. Add Unit/Integration Test Around Change Detection
+**Owner:** Naren  
+**GitHub Issue:** [#122](https://github.com/Georgia-Southwestern-State-Univeristy/term-project-group-4/issues/122)
```
Comment (hmhawn) [inline docs/final/week13-sprint.md:79]: potentially needing updated if the test for #122 is changed

**Review by hmhawn — COMMENTED** [review]
I only see two things worth tightening:

Items 7 and 8 feel very similar (both around auth error testing). It might be cleaner to either combine them into a single backlog item or clarify the distinction more explicitly so the scope is clearer.
The acceptance criteria for Issue #122 doesn’t quite match the actual goal of change detection. Right now it describes updating a trip and verifying a success toast, which is more of a general workflow test. Since #122 is about change detection, the criteria should explicitly cover behavior like the Update button being disabled after load and only enabling after a real change...... also #122 may need to be updated if the open PR for this issue is modified based on feedback. If that pr is merged as is, changes here may be minimal

Other than that, this looks well-structured and aligned with the deliverable.

File: docs/final/week13-sprint.md (hunk 1 above)
Comment (raya311) [reply docs/final/week13-sprint.md:19]: 7 Tests: Error message shown, URL parameter removed
8 Tests: Form hidden, login button available, error message shown, URL cleaned

Better keep them separate for test result purposes. If we combine both, one failure mask other result.

Comment (hmhawn) [reply docs/final/week13-sprint.md:19]: that makes sense. thanks for clarifying.

**Review by hmhawn — APPROVED** [review]
This looks good to me now
